src/mumble/VolumeSliderWidgetAction.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "VolumeSliderWidgetAction.h"
+#include "VolumeAdjustment.h"
+
+#include <cmath>
+
+#include <QtWidgets/QSlider>
+#include <QtWidgets/qtooltip.h>
+
+VolumeSliderWidgetAction::VolumeSliderWidgetAction(QObject *parent) : QWidgetAction(parent) {
+	qsVolumeSliderWidgetAction = new QSlider(Qt::Horizontal);
+
+	qsVolumeSliderWidgetAction->setMinimum(-60);
+	qsVolumeSliderWidgetAction->setMaximum(30);
+	qsVolumeSliderWidgetAction->setAccessibleName(tr("Volume Adjustment"));
+	qsVolumeSliderWidgetAction->setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::MinimumExpanding);
+
+	blockSliderSignals(true);
+
+	connect(qsVolumeSliderWidgetAction, SIGNAL(valueChanged(int)), this, SLOT(on_VolumeSlider_valueChanged(int)));
+	connect(qsVolumeSliderWidgetAction, SIGNAL(sliderReleased()), this, SLOT(on_VolumeSlider_sliderReleased()));
```

Comment:
We should probably prefer the newer function pointer signature for connecting these slots. This has the benefit of compile errors instead of runtime issues in case the signal's and slot's signature doesn't match up.